src/builder.rs

```diff
@@ -348,25 +348,56 @@ impl<'ctx> Builder<'ctx> {
     }
 
     // TODOC: Heap allocation
-    pub fn build_malloc<T: BasicType<'ctx>>(&self, ty: T, name: &str) -> PointerValue<'ctx> {
+    pub fn build_malloc<T: BasicType<'ctx>>(&self, ty: T, name: &str) -> Result<PointerValue<'ctx>, &'static str> {
+        // LLVMBulidMalloc segfaults if ty is unsized
+        let is_sized = match ty.as_basic_type_enum() {
```

Comment:
I think we should add an `is_sized` method to the `BasicValue` trait so that this is a little more seamless

Reply:
That's a good idea. Since you put yourself as the assignee, I assume you plan on making the change.

Reply:
Whoops, didn't mean to assign myself

Reply:
In the interest of time, I can make the change though